Library/ViewModels/PledgeViewModel.swift

```diff
@@ -449,87 +446,115 @@ public class PledgeViewModel: PledgeViewModelType, PledgeViewModelInputs, Pledge
+      .map(first)
+      .ignoreValues()
 
-    let createApplePayBackingEventErrors = createApplePayBackingEvent.errors()
-      .map { $0.localizedDescription }
+    let didInitiateApplePayBacking = Signal.merge(
+      willCreateApplePayBacking,
+      willUpdateApplePayBacking
+    )
 
-    let createApplePayBackingError = createApplePayBackingEventErrors
-      .takeWhen(createApplePayBackingCompleted)
+    let paymentAuthorizationDidFinish = didInitiateApplePayBacking
+      .takeWhen(self.paymentAuthorizationDidFinishSignal)
 
-    self.createBackingError = Signal.merge(
-      createApplePayBackingError,
-      createBackingEventError
+    let updateCreateApplePayBackingCompleted = Signal.combineLatest(
+      Signal.merge(
+        updateBackingEvent.filter { $0.isTerminating }.ignoreValues(),
+        createBackingEvent.filter { $0.isTerminating }.ignoreValues()
+      ),
+      paymentAuthorizationDidFinish
     )
 
-    let applePayTransactionCompleted = Signal.combineLatest(project, createApplePayBackingEventSuccess)
-      .takeWhen(createApplePayBackingCompleted)
-      .map(first)
+    let updateCreateBackingEventValues = Signal.merge(
+      createBackingEvent.values().map { $0 as StripeSCARequiring },
+      updateBackingEvent.values().map { $0 as StripeSCARequiring }
+    )
 
-    let createBackingTransactionSuccess = project.takeWhen(createBackingEventSuccess)
+    let updateCreateBackingEventValuesNoSCA = updateCreateBackingEventValues
+      .filter(requiresSCA >>> isFalse)
 
-    self.goToThanks = Signal.merge(applePayTransactionCompleted, createBackingTransactionSuccess)
+    let updateCreateBackingDidCompleteNoSCA = submitButtonTapped
+      .takeWhen(updateCreateBackingEventValuesNoSCA)
+      .filter(isTrue)
+      .ignoreValues()
 
-    // MARK: - Success/Failure Update
+    let updateCreateBackingEventValuesRequiresSCA = updateCreateBackingEventValues
+      .filter(requiresSCA)
 
-    let updatePaymentAuthorizationDidFinishSignal = willUpdateApplePayBacking
-      .takeWhen(self.paymentAuthorizationDidFinishSignal)
+    self.beginSCAFlowWithClientSecret = updateCreateBackingEventValuesRequiresSCA.map { $0.clientSecret }
+      .skipNil()
 
-    let updateApplePayBackingCompleted = Signal.combineLatest(
-      updateBackingEvent
-        .filter { $0.isTerminating }
-        .ignoreValues(),
-      updatePaymentAuthorizationDidFinishSignal
-    )
+    let didCompleteApplePayBacking = updateCreateBackingEventValues
+      .takeWhen(updateCreateApplePayBackingCompleted)
 
-    let updateApplePayBackingSuccess = updateBackingEvent.values()
-      .takeWhen(updateApplePayBackingCompleted)
+    let creatingContext = context.filter { $0.isCreating }
 
-    let updateBackingDidCompleteApplePay = updateApplePayBackingSuccess
-      .ignoreValues()
+    #warning("To be removed once new CreateBacking is implemented to handle Apple Pay")
+    let deprecatedCreateApplePayBackingCompleted = Signal.combineLatest(
```

Comment:
Self-explanatory but this should be removed once we're handling Apple Pay through `CreateBacking`.